python/mxnet/image/image.py

```diff
@@ -308,152 +430,481 @@ def random_size_crop(src, size, min_area, ratio, interp=2):
-            """Augmenter body"""
-            alpha = 1.0 + random.uniform(-contrast, contrast)
-            gray = src * coef
-            gray = (3.0 * (1.0 - alpha) / gray.size) * nd.sum(gray)
-            src *= alpha
-            src += gray
-            return [src]
-
-        ts.append(caug)
-
-    if saturation > 0:
-        def saug(src):
-            """Augmenter body"""
-            alpha = 1.0 + random.uniform(-saturation, saturation)
-            gray = src * coef
-            gray = nd.sum(gray, axis=2, keepdims=True)
-            gray *= (1.0 - alpha)
-            src *= alpha
-            src += gray
-            return [src]
-
-        ts.append(saug)
-    return RandomOrderAug(ts)
-
-
-def LightingAug(alphastd, eigval, eigvec):
-    """Add PCA based noise."""
-
-    def aug(src):
+
+class BrightnessJitterAug(Augmenter):
+    """Random brightness jitter augmentation.
+
+    Parameters
+    ----------
+    brightness : float
+        The brightness jitter ratio range, [0, 1]
+    min_val : float, default=0
+        Minimum value after jittering
+    max_val : float ,default=255
+        Maximum value after jittering
+    """
+    def __init__(self, brightness, min_val=0, max_val=255):
+        super(BrightnessJitterAug, self).__init__(brightness=brightness)
+        self.brightness = brightness
+        self.min_val = min_val
+        self.max_val = max_val
+
+    def __call__(self, src):
+        """Augmenter body"""
+        alpha = 1.0 + random.uniform(-self.brightness, self.brightness)
+        src *= alpha
+        return [clip_image(src, self.min_val, self.max_val)]
+
+
+class ContrastJitterAug(Augmenter):
+    """Random contrast jitter augmentation.
+
+    Parameters
+    ----------
+    contrast : float
+        The contrast jitter ratio range, [0, 1]
+    min_val : float, default=0
+        Minimum value after jittering
+    max_val : float ,default=255
+        Maximum value after jittering
+    """
+    def __init__(self, contrast, min_val=0, max_val=255):
+        super(ContrastJitterAug, self).__init__(contrast=contrast)
+        self.contrast = contrast
+        self.coef = nd.array([[[0.299, 0.587, 0.114]]])
+        self.min_val = min_val
+        self.max_val = max_val
+
+    def __call__(self, src):
+        """Augmenter body"""
+        alpha = 1.0 + random.uniform(-self.contrast, self.contrast)
+        gray = src * self.coef
+        gray = (3.0 * (1.0 - alpha) / gray.size) * nd.sum(gray)
+        src *= alpha
+        src += gray
+        return [clip_image(src, self.min_val, self.max_val)]
+
+
+class SaturationJitterAug(Augmenter):
+    """Random saturation jitter augmentation.
+
+    Parameters
+    ----------
+    saturation : float
+        The saturation jitter ratio range, [0, 1]
+    min_val : float, default=0
+        Minimum value after jittering
+    max_val : float ,default=255
+        Maximum value after jittering
+    """
+    def __init__(self, saturation, min_val=0, max_val=255):
```

Comment:
@piiswrong I've noticed bad visual artifacts if we don't clip it before normalization. Clip to 0,255 make it more like a natural image.